src/vsphere_cpi/spec/unit/cloud/vsphere/nsxt_provider_spec.rb

```diff
@@ -633,4 +634,43 @@
       nsxt_provider.remove_vm_from_server_pools(vm_ip_address)
     end
   end
+
+  describe '#retrieve_all_ns_groups_with_pagination' do
+    let(:nsgroup_1) do
+      NSXT::NSGroup.new(:id => 'id-1', :display_name => 'test-nsgroup-1')
+    end
+    let(:nsgroup_2) do
+      NSXT::NSGroup.new(:id => 'id-2', :display_name => 'test-nsgroup-2')
+    end
+    let(:nsgroup_3) do
+      NSXT::NSGroup.new(:id => 'id-3', :display_name => 'test-nsgroup-3')
+    end
+    let(:nsgroup_4) do
+      NSXT::NSGroup.new(:id => 'id-4', :display_name => 'test-nsgroup-4')
+    end
+
+    context 'whe there is only one page' do
+      let(:result_list_1) { double(Object, results: [nsgroup_1, nsgroup_2], cursor: nil) }
+      it 'returns all NS Groups in the page' do
+        allow_any_instance_of(VSphereCloud::NSXTProvider).to receive(:grouping_obj_svc).and_return(grouping_obj_svc)
+        allow(grouping_obj_svc).to receive(:list_ns_groups).and_return(result_list_1)
+        objects_result = nsxt_provider.send(:retrieve_all_ns_groups_with_pagination)
+        expect(objects_result.size).to eq(2)
+      end
+    end
+
+    context 'when there are two pages' do
+      let(:result_list_1) { double(Object, results: [nsgroup_1, nsgroup_2], cursor: 1) }
+      let(:result_list_2) { double(Object, results: [nsgroup_3, nsgroup_4], cursor: nil) }
+      let(:result_list_1_cursor) { "fake-result-list-1-cursor"}
+      it 'returns all NS Groups in the two pages' do
+        allow_any_instance_of(VSphereCloud::NSXTProvider).to receive(:grouping_obj_svc).and_return(grouping_obj_svc)
+        allow(grouping_obj_svc).to receive(:list_ns_groups).and_return(result_list_1)
+        allow(grouping_obj_svc).to receive(:list_ns_groups).with(anything).and_return(result_list_2)
```

Comment:
Can you also put a expect 

`expect(grouping_obj_svc).to receive(:list_ns_groups).with(any_args).twice`

Reply:
Done.